frontend/src/bundles/common/types/button.type.ts

```diff
@@ -0,0 +1,28 @@
+import { type ReactNode } from 'react';
+
+enum ButtonVariant {
+    Primary = 'primary',
+    Secondary = 'secondary',
+    Tertiary = 'tertiary',
+}
+
+enum ButtonSize {
+    Small = 'small',
+    Medium = 'medium',
+}
+
+type ButtonType = 'button' | 'submit';
+
+type ButtonProperties = {
+    size: ButtonSize;
+    variant: ButtonVariant;
+    label: string;
+    leftIcon?: ReactNode;
+    rightIcon?: ReactNode;
+    type?: ButtonType;
+    isDisabled?: boolean;
+    className?: string;
+    onClick?: () => void;
+};
+
+export { type ButtonProperties, ButtonSize, ButtonVariant };
```

Comment:
Move these types to the component file and re-export from components.ts. Switch the enum to an `as const `object. Make sure the enum keys are in uppercase.